docs/enhancements/infrastructure-monitoring-service-api/lifecycle.md

```diff
@@ -0,0 +1,530 @@
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarms/{alarmEventRecordId}`              | GET             | Retrieve exactly one alarm identified by `alarmEventRecordId`.      | None                                                                         | Exactly one `AlarmEventRecord`      |
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarms/{alarmEventRecordId}`              | PATCH           | Modify exactly one alarm identified by `alarmEventRecordId` to ack. | `AlarmEventRecordModifications`(no perceivedSeverity only alarmAcknowledged) | `AlarmEventRecordModifications`     |
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions`                       | GET             | Retrieve the list of alarm subscriptions.                           | None                                                                         | A list of `AlarmSubscriptionInfo`   |
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions`                       | POST            | Create a new alarm subscriptions.                                   | `AlarmSubscriptionInfo`                                                      | Exactly one `AlarmSubscriptionInfo` |
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions/{alarmSubscriptionId}` | GET             | Retrieve exactly one subscription using `alarmSubscriptionId`.      | None                                                                         | Exactly one `AlarmSubscriptionInfo` |
+| `/O2ims_infrastructureMonitoring/{apiVersion}/alarmSubscriptions/{alarmSubscriptionId}` | DELETE          | Delete exactly one subscription using `alarmSubscriptionId`.        | None                                                                         | None                                |
+
+
+
+| **Internal Endpoint**                           | **HTTP Method** | **Description**                               | **Input Payload**                                                        | **Returned Data** |
+|-------------------------------------------------|-----------------|-----------------------------------------------|--------------------------------------------------------------------------|-------------------|
+| `internal/v1/caas-alerts/alertmanager`          | POST            | Alertmanager notifications comes through here | https://prometheus.io/docs/alerting/latest/configuration/#webhook_config | None              |
+| `internal/v1/hardware-alerts/{hw-vendor-name}}` | POST            | TBD                                           | TBD                                                                      | TBD               |
+
+
+
+### `alarms` family
+
+### Steps for `/O2ims_infrastructureMonitoring/{apiVersion}/alarms` with GET
+1. Get active alarms from `alarm_event_record` table
```

Comment:
2. we need apply the filter via DB select schemas

Reply:
Maybe Im missing something here....this is simply a get query on the DB with no query parameters. Where is the filter coming from? 

Essentially get all AlarmEventRecord right?

Reply:
I will check the recording by Daniel

Reply:
Added doc for `?filter`